libmisc/copydir.c

```diff
@@ -522,14 +522,13 @@ static int copy_dir (const struct path_info *src, const struct path_info *dst,
 		return -1;
 	}
 #endif				/* WITH_SELINUX */
-	if (   (mkdirat (dst->dirfd, dst->name, statp->st_mode) != 0)
+	if (   (mkdirat (dst->dirfd, dst->name, 0700) != 0)
 	    || (chownat_if_needed (dst, statp,
 	                         old_uid, new_uid, old_gid, new_gid) != 0)
+	    || (chmod (dst, statp->st_mode) != 0)
```

Comment:
This chmod call needs to be fixed, similar to https://github.com/shadow-maint/shadow/pull/565

Reply:
I have adjusted this pull request now. The other one was strictly about correcting regressions. This one is strictly about fixing the permissions. To keep these pull requests separated by topic, I had to accept that they contain this merge conflict. But thank you for pointing out, so it wouldn't be missed!